ui/desktop/src/components/settings/providers/modal/ProviderConfiguationModal.tsx

```diff
@@ -53,6 +56,95 @@ export default function ProviderConfigurationModal() {
       setValidationErrors({});
       setShowDeleteConfirmation(false);
       setIsActiveProvider(false); // Reset active provider state
+
+      // If this is a custom provider, fetch the full provider JSON to pre-fill extra fields
+      if (currentProvider.name.startsWith('custom_')) {
+        (async () => {
+          try {
+            const secretKey = await window.electron.getSecretKey();
+            // Try generated API client first
+            try {
+              /* eslint-disable @typescript-eslint/no-explicit-any */
+              const clientMod = await import('../../../../api/client.gen');
+              const client = (
+                clientMod as unknown as { client?: { get?: (...args: any[]) => Promise<any> } }
+              ).client;
+              if (client && typeof client.get === 'function') {
+                const resp = await client.get({
+                  url: '/config/custom-providers/{id}',
+                  path: { id: currentProvider.name },
+                  headers: { 'X-Secret-Key': secretKey },
+                });
+                const body =
+                  (resp as unknown as any)?.data ?? (resp as unknown as any)?.body ?? resp;
+                if (body) {
+                  setConfigValues((prev) => ({
+                    ...prev,
+                    display_name: body.display_name ?? prev.display_name,
+                    description: body.description ?? prev.description,
+                    headers: body.headers ? JSON.stringify(body.headers) : prev.headers,
+                    timeout_seconds: body.timeout_seconds
+                      ? String(body.timeout_seconds)
+                      : prev.timeout_seconds,
+                    supports_streaming:
+                      body.supports_streaming !== undefined
+                        ? body.supports_streaming
+                        : prev.supports_streaming,
+                  }));
+                }
+                return;
+              }
+              /* eslint-enable @typescript-eslint/no-explicit-any */
+            } catch (e) {
+              console.debug('API client get failed, falling back to fetch', e);
+            }
+
+            // Fallback to a direct fetch
```

Comment:
why the fallback? is it because client wasn't initialized with the right endpoint? that should I think be working better now as of https://github.com/block/goose/pull/4338 and we shouldn't need this